File: OC/Tools_OC/ViewController.m
//
//  ViewController.m
//  Tools_OC
//
//  Created by Jalen.He on 2017/6/16.
//  Copyright © 2017年 Jalen. All rights reserved.
//

#import "ViewController.h"
#import <CommonCrypto/CommonCrypto.h>
#import "NSString+Encryption.h"
#import "GTMBase64.h"

#import "UILabel+Extension.h"
#import "UIBarButtonItem+Extension.h"
#import "UIButton+Extension.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self creatUI];
    
//    [self action];
}

- (void)creatUI {
    UIButton *btn = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 90, 90)];
    btn.center = self.view.center;
    [btn setTitle:@"Action" forState:UIControlStateNormal];
    [btn setImage:[UIImage imageNamed:@"icon_C_1"] forState:UIControlStateNormal];
    btn.titleLabel.textColor = [UIColor whiteColor];
    btn.backgroundColor = [UIColor darkGrayColor];
    btn.layer.cornerRadius = 5.0;
    [btn addTarget:self action:@selector(action) forControlEvents:UIControlEventTouchUpInside];
    
    [btn leftTitleAndRightImageFits];
//    [btn topImageAndBottomTitleFits:16.0];
    
    [self.view addSubview:btn];
    
    UIBarButtonItem *leftItem = [[UIBarButtonItem alloc] initWithTitle:@"Back" tintColor:[UIColor darkGrayColor] target:self action:@selector(action)];
    self.navigationItem.leftBarButtonItem = leftItem;
    
    UIBarButtonItem *rightItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"icon_C_1"] target:self action:@selector(action)];
    self.navigationItem.rightBarButtonItem = rightItem;
}

- (void)action {
    NSLog(@"------ Normal action");
}


@end

File: Swift/Tools/ViewController.swift
//
//  ViewController.swift
//  Tools
//
//  Created by 杨方明 on 17/3/15.
//  Copyright © 2017年 杨方明. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.initUI()
    }
    
    private func initUI() {
        self.view.backgroundColor = .white
        
        colorView.frame = CGRect(x: 0, y: 0, width: 300, height: 300)
        colorView.center = self.view.center
        self.view.addSubview(colorView)
        
        let actionBtn = UIButton(frame: CGRect(x: 0, y: 0, width: 90, height: 90))
        actionBtn.center = self.view.center
        actionBtn.setImage(UIImage(named: "icon_C_1"), for: .normal)
        actionBtn.setTitle("Action", for: .normal)
        
        actionBtn.titleLabel?.textColor = UIColor.white
        actionBtn.backgroundColor = UIColor.darkGray
        actionBtn.layer.cornerRadius = 5.0
        actionBtn.addTarget(self, action: #selector(test), for: .touchUpInside)
        
        self.view.addSubview(actionBtn)
        
        let leftItem = UIBarButtonItem(title: "Back", tintColor: .darkGray, target: self, action: #selector(action))
        self.navigationItem.leftBarButtonItem = leftItem
        
        let rightItem = UIBarButtonItem(image: UIImage(named: "icon_C_1"), target: self, action: #selector(action2))
        self.navigationItem.rightBarButtonItem = rightItem
    }
    
    @objc func test() {
        colorView.backgroundColor = UIColor.jlt.randomColor
    }
    
    @objc func action(btn: UIButton) {
        colorView.backgroundColor = UIColor.jlt.randomColorDeep
    }
    
    @objc func action2(btn: UIButton) {
        colorView.backgroundColor = UIColor.jlt.randomColorLight
    }
    
    
    private lazy var colorView: UIView = {
        return UIView()
    }()
}

